internal/factsengine/gathering.go

```diff
@@ -60,10 +72,12 @@ func gatherFacts(
 			}
 
 			return nil
+
 		})
 	}
 
-	if err := g.Wait(); err != nil {
+	err := g.Wait()
+	if err != nil {
```

Comment:
nitpick: why do you remove the one liner? doesn't it look better?

Reply:
Unintentional, restored in https://github.com/trento-project/agent/pull/360/commits/c66e25b4bf909fb4aebf7830690e27b186833970